rust/worker/src/execution/operators/merge_metadata_results.rs

```diff
@@ -96,150 +105,327 @@ impl Operator<MergeMetadataResultsOperatorInput, MergeMetadataResultsOperatorOut
             input.record_segment_definition.id.to_string()
         );
 
-        let mut ids: Vec<String> = Vec::new();
-        let mut metadata = Vec::new();
-        let mut documents = Vec::new();
-        // Add the data from the brute force results
-        for (log_entry, index) in input.filtered_log.iter() {
-            ids.push(log_entry.record.id.to_string());
-            let output_metadata = match &log_entry.record.metadata {
-                Some(log_metadata) => match update_metdata_to_metdata(log_metadata) {
-                    Ok(metadata) => Some(metadata),
-                    Err(e) => {
-                        println!("Error converting log metadata: {:?}", e);
-                        return Err(MergeMetadataResultsOperatorError::MetadataConversionError(
-                            e,
-                        ));
+        let mut merged_offset_ids: Option<HashSet<u32>> = match &input.user_offset_ids {
+            Some(user_ids) => {
+                match &input.filtered_offset_ids {
+                    // Intersect with ids filtered from the where clause.
+                    Some(filtered_ids) => {
+                        let merged_vecs =
+                            merge_sorted_vecs_conjunction(user_ids.clone(), filtered_ids.clone());
```

Comment:
Why do these methods need ownership over the vecs? That doesn't make sense to me. They should borrow slices.

cc @beggers

Reply:
I think they could borrow slices

Reply:
thank you!